File: scripts/check-black-bags.mjs
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

// Load .env.local from parent directory
dotenv.config({ path: join(__dirname, '..', '.env.local') });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('Missing Supabase credentials');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkBlackBags() {
  // Check template
  const { data: templates, error: tError } = await supabase
    .from('task_templates')
    .select('id, title, window_start_local, window_end_local, kind, duration_minutes, priority, repeat_unit, repeat_interval, is_appointment, is_routine')
    .ilike('title', '%black bags%');

  if (tError) {
    console.error('Template error:', tError);
    return;
  }

  console.log('\n=== BLACK BAGS TEMPLATE ===');
  console.log(JSON.stringify(templates, null, 2));

  if (!templates || templates.length === 0) {
    console.log('No Black bags template found');
    return;
  }

  // Check today's scheduled task
  const today = new Date().toISOString().split('T')[0];
  const { data: scheduled, error: sError } = await supabase
    .from('scheduled_tasks')
    .select('title, start_time, end_time, local_date, is_appointment, is_routine')
    .eq('local_date', today)
    .ilike('title', '%black bags%');

  if (sError) {
    console.error('Scheduled error:', sError);
    return;
  }

  console.log('\n=== BLACK BAGS SCHEDULED FOR TODAY ===');
  console.log(JSON.stringify(scheduled, null, 2));
}

checkBlackBags();

File: check-black-bags.py
from supabase import create_client

sb = create_client('https://prloxvewcsxaptzgxvyy.supabase.co', 'sb_secret_ksC74t4wtkeOcdvNtxlVCQ_hkygZocg')

result = sb.table('scheduled_tasks')\
    .select('title, start_time, end_time, duration_minutes, is_completed')\
    .eq('local_date', '2025-12-02')\
    .eq('user_id', '3c877140-9539-47b9-898a-45eeab392e39')\
    .eq('title', 'Black bags')\
    .execute()

print('Black bags on Dec 2:')
if result.data:
    for r in result.data:
        print(f'  Title: {r["title"]}')
        print(f'  Start time: {r.get("start_time")}')
        print(f'  End time: {r.get("end_time")}')
        print(f'  Duration: {r.get("duration_minutes")} minutes')
        print(f'  Completed: {r.get("is_completed")}')
else:
    print('  No Black bags task found for Dec 2')
